### ISSUE TITLE
🚀 Feature: A script to revert changes made via GitHub API

### ISSUE THREAD

**Issue by danvk** [JoshuaKGoldberg/create-typescript-app#1379]
### Bug Report Checklist

- [X] I have pulled the latest `main` branch of the repository.
- [X] I have [searched for related issues](https://github.com/JoshuaKGoldberg/create-typescript-app/issues?q=is%3Aissue) and found none that matched my issue.

### Overview

I ran `npx create-typescript-app` on https://github.com/danvk/literate-ts and created a PR, but I didn't merge it. I was surprised to see that changes had been made to my repo via the GitHub API (branch protection rules) independently of that PR.

I'll manually revert them, but it would be nice if these changes could be made / reverted more easily via a script.

### Additional Info

Would likely be fixed by #1309. Feel free to close this as a dupe if that's a feasible solution.

**Comment by JoshuaKGoldberg** [comment]
Yeah it's annoying 😕 I've been inconvenienced by this before. It would be nice if the previous state of things could be stored somewhere.

Proposal: how about adding something to the output at the end of `--mode migrate`? It should:
* indicate explicitly that GitHub settings were made
* show the command to run a script that sets values back to what they were 

...where the script is a file written to disk?

> Would likely be fixed by #1309

I think this will still be an issue there? Since you might not want to preserve rulesets changes.

**Comment by johnnyreilly** [comment]
>  was surprised to see that changes had been made to my repo via the GitHub API (branch protection rules) independently of that PR.

Just to satisfy my own curiosity, what was the nature of the changes made? And to which branch?

**Comment by JoshuaKGoldberg** [comment]
Branch protections get set up on the `main` branch no matter what:

https://github.com/JoshuaKGoldberg/create-typescript-app/blob/3a8c7c03723eca5a50998270b75f15a7998ca69c/src/steps/initializeGitHubRepository/initializeBranchProtectionSettings.ts#L12

**Comment by johnnyreilly** [comment]
"We will protect you whether you like it or not!"

**Comment by danvk** [comment]
I wound up deleting my `main` branch protection and recreating it as best I could remember.

> > Would likely be fixed by #1309
> 
> I think this will still be an issue there? Since you might not want to preserve rulesets changes.

Right, I misunderstood this GitHub feature. I thought it let you configure rulesets via code, which really would solve the problem.

Still, I wonder if you could create a new ruleset and let the user delete whatever existing branch protections they had before once they're ready to commit. At least then the new ruleset wouldn't be overriding anything.
